main.nf

```diff
@@ -0,0 +1,932 @@
+        c_r1 = params.clip_r1 > 0 ? "--clip_r1 ${params.clip_r1}" : ''
+        c_r2 = params.clip_r2 > 0 ? "--clip_r2 ${params.clip_r2}" : ''
+        tpc_r1 = params.three_prime_clip_r1 > 0 ? "--three_prime_clip_r1 ${params.three_prime_clip_r1}" : ''
+        tpc_r2 = params.three_prime_clip_r2 > 0 ? "--three_prime_clip_r2 ${params.three_prime_clip_r2}" : ''
+        if (params.singleEnd) {
+            """
+            trim_galore --gzip $c_r1 $tpc_r1 $reads
+            """
+        } else {
+            """
+            trim_galore --paired --gzip $c_r1 $c_r2 $tpc_r1 $tpc_r2 $reads
+            """
+        }
+    }
+}
+
+
+/*
+ * STEP 2 - Map with BWA Mem
+ */
+
+process bwamem {
+    tag "$name"
+
+    input:
+    set val(name), file(reads) from trimmed_reads
+    file(bwa_index) from bwa_index
+
+    output:
+    set val(name), file("${name}_bwa.sam") into raw_aln_sam
+    file '.command.log' into bwa_stdout
+
+
+    script:
+    rg="\'@RG\\tID:${params.run_id}\\tSM:${params.run_id}\\tPL:illumina\'"
+    if(params.singleEnd){
+    """
+         bwa mem \\
+        -R $rg \\
+        -t ${task.cpus} \\
+        -k 2 \\
+        $params.gfasta \\
+        $reads \\
+            > ${name}_bwa.sam
+
+    # Print version number to standard out
+    echo "BWA Version:"\$(bwa 2>&1)
+    """
+    } else {
+    """
+        bwa mem \\
+        -R $rg \\
+        -t ${task.cpus} \\
+        -k 2 \\
+        $params.gfasta \\
+        $reads\\
+            > ${name}_bwa.sam
+
+    # Print version number to standard out
+    echo "BWA Version:"\$(bwa 2>&1)
+    """
+
+}
+}
+
+/*
+*  STEP 3 - Convert to BAM, sort BAM  
+*/ 
+
+process sortSam {
+    tag "${name}"
+    publishDir "${params.outdir}/BWAmem", mode: 'copy',
+        saveAs: {filename -> params.saveAlignedIntermediates ? "$filename" : null }
+    
+    input:
+    set val(name), file(raw_sam) from raw_aln_sam
+    
+
+    output: 
+    set val(name), file("${raw_sam}.sorted.bam") into samples_sorted_bam
+
+    script:
+    def avail_mem = task.memory == null ? '' : "-m ${Math.floor(task.memory.toBytes() / task.cpus)}"
+    """
+    samtools sort \\
+        $raw_sam \\
+        -@ ${task.cpus} $avail_mem \\
+        -o ${raw_sam}.sorted.bam
+    """
+}
+
+/*
+*  STEP 4 - Mark PCR duplicates in sorted BAM file
+*/ 
+
+process markDuplicates {
+    tag "${name}"
+    publishDir "${params.outdir}/markDup/metrics", mode: 'copy',
+        saveAs: { filename -> filename.indexOf(".dup_metrics") > 0 ? filename : null }
+
+    input:
+    set val(name), file(sorted_bam) from samples_sorted_bam
+
+    output:
+    set val(name), file("${name}_markdup.bam") into samples_markdup_bam
+    file("${name}.dup_metrics") into markdup_results
+    file '.command.log' into markDuplicates_stdout
+
+    script:
+    """
+        picard MarkDuplicates \\
+        INPUT=$sorted_bam \\
+        OUTPUT=${name}_markdup.bam \\
+        METRICS_FILE=${name}.dup_metrics \\
+        TMP_DIR=tmp \\
```

Comment:
Is it not better to leave this as default?

Reply:
I removed the statement ... in 7565a44465dec86dd333e15595235d2b1c3bacce